lib/modules/modules.go

```diff
@@ -74,6 +75,8 @@ type Modules interface {
 	Features() Features
 	// BuildType returns build type (OSS or Enterprise)
 	BuildType() string
+	// ValidateResource performs additional resource checks.
+	ValidateResource(types.Resource) error
```

Comment:
WDYT about making this a top-level function?
Something like:
```go
func ValidateResource(r types.Resource) error {
  if !GetModules().Features().Cloud {
    return nil
  }
  // validation goes here...
}
```

That way we don't need to separately update `teleport.e`

Reply:
To satisfy the current requirements a top-level function would suffice. I'm not sure it completely eliminates the need for `teleport.e` updates though -- it depends on how "tight" we want our tests to be. If the resource checks should be tested with `enterpriseModules` directly no matter what, the tests will still have to go to `teleport.e`.

Reply:
You can populate the `Cloud` flag with:
```
type cloudModules struct {
  *defaultModules
}

func (m cloudModules) Features() Features {
  f := m.defaultModules.Features()
  f.Cloud = true
  return f
}
```

Reply:
@awly Converted `ValidateResource` into a top-level function. I had to do additional adjustments to `modules_test` due to cyclic imports, let me know if it's acceptable.